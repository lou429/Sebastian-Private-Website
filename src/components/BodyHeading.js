import Helmet from 'react-helmet';

function getTitle(heading) {
    let start = "Sebastian's ";
    return start + heading.toLowerCase();
}

function getContent(heading) {
    return heading + "-no-content";
}

function BodyHeading(props) {
    return(
        <div className="content-heading">
            <Helmet>
                <title>{getTitle(props.heading)}</title>
                <meta name="description" content={getContent(props.heading)}/>
            </Helmet>
            <h1>{props.heading}</h1>
        </div>
    );
}

export default BodyHeading;