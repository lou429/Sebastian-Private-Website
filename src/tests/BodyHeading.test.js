import BodyHeading from '../components/BodyHeading';
import { render } from '@testing-library/react';

test('Heading content can be rendered', () => {
    let text = "Test heading"; 

    const { container, getByText } = render(<BodyHeading heading={text}/>);
    
    expect(getByText(text)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});
