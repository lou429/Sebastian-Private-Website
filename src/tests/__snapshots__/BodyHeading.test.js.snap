// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading content can be rendered 1`] = `
<div>
  <div
    class="content-heading"
  >
    <h1>
      Test heading
    </h1>
  </div>
</div>
`;
